### PR TITLE
enh: support setting compatible software name

### PR DESCRIPTION
**Pull request by YamingPei**
Supports setting compatible software in the odbc data source Manager interface

### CODE REVIEW

File: src/inc/conn.h [hunk 1]
```diff
@@ -36,6 +36,10 @@ EXTERN_C_BEGIN
 
 void conn_cfg_release(conn_cfg_t *conn_cfg) FA_HIDDEN;
 
+const custprod_case_t *conn_get_custprod_case_by_index(size_t index) FA_HIDDEN;
+
+const custprod_case_t *conn_get_custprod_case_by_name(const char *s, size_t n) FA_HIDDEN;
```
Comment (sheyanjie-qq) [inline src/inc/conn.h:41]: why use  case in func name?

File: src/core/conn.c [hunk 2]
```diff
@@ -62,32 +62,43 @@ void conn_cfg_release(conn_cfg_t *conn_cfg)
...
-      conn_cfg->customproduct = _cases[i].customproduct;
-      return 0;
+      return &custprod_cases[i];
     }
   }
+  return NULL;
+}
 
-  return -1;
+int conn_cfg_set_customproduct(conn_cfg_t *conn_cfg, const char *s, size_t n)
```
Comment (sheyanjie-qq) [inline src/core/conn.c:91]: conn_cfg_set_custom_product

File: src/core/conn.c [hunk 3]
```diff
@@ -62,32 +62,43 @@ void conn_cfg_release(conn_cfg_t *conn_cfg)
   memset(conn_cfg, 0, sizeof(*conn_cfg));
 }
 
-int conn_cfg_set_customproduct(conn_cfg_t *conn_cfg, const char *s, size_t n)
+const custprod_case_t custprod_cases[] = {
```
Comment (sheyanjie-qq) [inline src/core/conn.c:65]: custprod_list  or  custprod_array

File: src/inc/conn.h (hunk 1 above)
Comment (YamingPei) [reply src/inc/conn.h]: I removed the `case` from the function name

**Pull request merged**